stablehlo/transforms/StablehloLegalizeQuantToMath.cpp

```diff
@@ -153,7 +151,21 @@ Type getQuantStorageType(Type type) {
 
   auto quantizedType = getQuantType(type);
   if (succeeded(quantizedType)) {
-    return getQuantStorageType(*quantizedType);
+    return quantizedType->getStorageType();
+  }
+  return type;
+}
+
+// Extracts expressed type of a UQ type. Return original type if it is no UQ
+// type.
+Type getQuantExpressedType(Type type) {
+  if (auto shaped = dyn_cast<ShapedType>(type)) {
```

Comment:
Any test case to exercise the newly supported behavior?

Reply:
tests in child PR

Reply:
Actually this new behavior is exercised  for the existing operations https://github.com/openxla/stablehlo/blob/b056d3da25c18169c14018a1e953112852a53453/stablehlo/transforms/StablehloLegalizeQuantizedOpToQDQ.cpp#L127 in which cases  `get-expressed type` is explicitly computing the return type for them (instead of auto-deriving those types: old-behavior).